docs/rules/no-commonjs.md

```diff
@@ -21,6 +21,24 @@ module.exports = { a: "b" }
 exports.c = "d"
 ```
 
+### Allow require
+
+If `allow-require` is provided as an option, `require` calls are valid:
+
+```js
+/*eslint no-commonjs: [2, "allow-require"]*/
```

Comment:
Let's use an object schema instead, so it's extensible - like `{ allowRequire: true, allowPrimitiveModules: true }`.

We should move away from string schemas (but continue to support the existing ones, ofc)